File: python/2/main.py
import sys


def sum_of_even_fibonacci_terms(limit: int) -> int:
    result = 0
    prev = 1
    current = 1
    while current < limit:
    	old_current = current
    	current = current + prev
    	prev = old_current
    	if current % 2 == 0:
    		result += current
    return result


def time_test():
    result = None
    for _ in range(100_000):
        result = sum_of_even_fibonacci_terms(4_000_000)
    print(result)


if __name__ == "__main__":
    if len(sys.argv) > 1: 
        time_test()
    else:
        print(sum_of_even_fibonacci_terms(4_000_000))

File: cpp/2/main.cpp
#include <iostream>

int sum_of_even_fibonacci_terms(int limit) {
	int result = 0;
	int prev = 1;
	int current = 1;
	int old_current = 0;
	while (current < limit) {
		old_current = current;
		current += prev;
		prev = old_current;
		if (current % 2 == 0) {
			result += current;
		}
	}
	return result;
}

void time_test() {
	long int result = 0;
	for (int i = 0; i < 100'000; i++) {
		result = sum_of_even_fibonacci_terms(4'000'000);
	}
	std::cout << result << '\n';
}

int main(int argc, char** argv) {
	if (argc > 1) {
		time_test();
	}
	else {
		std::cout << sum_of_even_fibonacci_terms(4'000'000) << '\n';
	}
	return 0;
}
